consul/src/main/java/com/linecorp/armeria/server/consul/ConsulUpdatingListenerBuilder.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.consul;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.net.URI;
+import java.time.Duration;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.client.Endpoint;
+import com.linecorp.armeria.common.HttpMethod;
+import com.linecorp.armeria.common.SessionProtocol;
+import com.linecorp.armeria.internal.consul.ConsulClientBuilder;
+import com.linecorp.armeria.server.Server;
+
+/**
+ * Builds a new {@link ConsulUpdatingListener}, which registers the server to Consul cluster.
+ * <h3>Examples</h3>
+ * <pre>{@code
+ * ConsulUpdatingListener listener = ConsulUpdatingListener.builder("myService")
+ *                                                         .consulPort(8501")
+ *                                                         .build();
+ * ServerBuilder sb = Server.builder();
+ * sb.serverListener(listener);
+ * }</pre>
+ */
+public final class ConsulUpdatingListenerBuilder extends ConsulClientBuilder {
+
+    private static final long DEFAULT_CHECK_INTERVAL_MILLIS = 10_000;
+    private final String serviceName;
+
+    @Nullable
+    private Endpoint serviceEndpoint;
+    @Nullable
+    private URI checkUri;
+    private String checkInterval = DEFAULT_CHECK_INTERVAL_MILLIS + "ms";
+    @Nullable
+    private HttpMethod checkMethod;
```

Comment:
Does it make sense to use `HttpMethod.HEAD` as the default?

Reply:
Yes, it makes sense. I tested this and applied it.